frontend/src/metabase-lib/lib/Dimension.js

```diff
@@ -821,6 +821,8 @@ export class ExpressionDimension extends Dimension {
   }
 }
 
+const UNAGGREGATED_SPECIAL_TYPES = new Set([TYPE.FK, TYPE.PK]);
```

Comment:
Might be worth just a small comment here explaining why FK and PK are special in this case per the PR discussion.